src/features/problems/queries/getProblem.ts

```diff
@@ -53,14 +52,13 @@ export const getProblem = async (
       [`getProblem-${code}`],
       { revalidate: false },
     )();
+    return {
+      ok: true,
+      data: data,
+    };
   } catch (e) {
-    if (e instanceof PrismaClientKnownRequestError) {
-      console.error(e.message);
-    } else {
-      console.error(e);
-    }
-    throw new Error("Internal Server Error");
+    return handleCatch(e);
   }
 };
 
-export const getMemoProblem = cache(getProblem);
+export const getProblemWithMemo = cache<typeof getProblem>(getProblem);
```

Comment:
Why do we need to specify the generic here? Shouldn't it be inferred?

Reply:
Agree